controllers/clusteroperator_controller_test.go

```diff
@@ -307,32 +312,137 @@ var _ = Describe("Component sync controller", func() {
 		Entry("Should provision AWS resources", testCase{
 			status: &configv1.InfrastructureStatus{
 				Platform: configv1.AWSPlatformType,
+				PlatformStatus: &configv1.PlatformStatus{
+					Type: configv1.AWSPlatformType,
+				},
 			},
 			featureGateSpec: externalFeatureGateSpec,
-			expected:        cloud.GetAWSResources(),
+			config: operatorConfig{
+				ManagedNamespace: testManagementNamespace,
+				ControllerImage:  "registry.ci.openshift.org/openshift:aws-cloud-controller-manager",
+			},
+			expected: cloud.GetAWSResources(),
 		}),
 		Entry("Should provision OpenStack resources", testCase{
 			status: &configv1.InfrastructureStatus{
 				Platform: configv1.OpenStackPlatformType,
+				PlatformStatus: &configv1.PlatformStatus{
+					Type: configv1.OpenStackPlatformType,
+				},
+			},
+			config: operatorConfig{
+				ManagedNamespace: testManagementNamespace,
+				ControllerImage:  "registry.ci.openshift.org/openshift:openstack-cloud-controller-manager",
 			},
 			featureGateSpec: externalFeatureGateSpec,
 			expected:        openstack.GetResources(),
 		}),
 		Entry("Should not provision resources for currently unsupported platform", testCase{
 			status: &configv1.InfrastructureStatus{
 				Platform: configv1.IBMCloudPlatformType,
+				PlatformStatus: &configv1.PlatformStatus{
+					Type: configv1.IBMCloudPlatformType,
+				},
 			},
 			featureGateSpec: externalFeatureGateSpec,
 		}),
 		Entry("Should not provision resources for AWS if external FeatureGate is not present", testCase{
 			status: &configv1.InfrastructureStatus{
 				Platform: configv1.AWSPlatformType,
+				PlatformStatus: &configv1.PlatformStatus{
+					Type: configv1.AWSPlatformType,
+				},
 			},
 		}),
 		Entry("Should not provision resources for OpenStack if external FeatureGate is not present", testCase{
 			status: &configv1.InfrastructureStatus{
 				Platform: configv1.OpenStackPlatformType,
+				PlatformStatus: &configv1.PlatformStatus{
+					Type: configv1.OpenStackPlatformType,
+				},
 			},
 		}),
 	)
 })
+
+func TestComposeConfig(t *testing.T) {
+	tc := []struct {
+		name          string
+		namespace     string
+		platform      configv1.PlatformType
+		imagesContent string
+		expectConfig  operatorConfig
+		expectError   bool
+	}{{
```

Comment:
i know it shouldn't matter, but i wonder if we should have a test with a duplicate entry just to make sure it doesn't break things?

Reply:
Added a test case for that in https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/29/files#diff-57ee0125add50ed49739258385c78a28c3d1a738e949eb9c1210653facdcb658R43-R52 Although we are testing json unmarshal here, let's still have it for clarity.